Get full dimensional polytope for the Python interface

hariszaf opened the pull request:
If applied, this commit will include a new class in the Python interface of _volesti_

The class called ```low_dim_HPolytope``` will allow the user:
* to build a low dimensional polytope
* to get its corresponding full dimensional polytope

For the latter, the ```full_dimensiolal_polytope``` function is used and its output includes in a tuple format the following:
* an object of the ```HPolytope``` class for the case of the full dimensional polytope; 
* the ```shift``` vector
* the ```N``` matrix

hariszaf committed:
Create setup_hz.py

hariszaf committed:
Create bindings_hz.cpp

hariszaf committed:
Create volestipy_hz.pyx

hariszaf committed:
Create bindings_hz.h

hariszaf committed:
Update bindings_hz.h

hariszaf committed:
Update bindings_hz.cpp

hariszaf committed:
Create volestipy_old.pyx

hariszaf committed:
Delete volestipy_hz.pyx

hariszaf committed:
Create bindings_old.cpp

hariszaf committed:
Delete bindings_hz.cpp

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Create bindings_old.h

hariszaf committed:
Delete bindings_hz.h

hariszaf committed:
Create setup_old.py

hariszaf committed:
Delete setup_hz.py

hariszaf committed:
Update setup.py

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.h

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update setup.py

hariszaf committed:
Update bindings.h

hariszaf committed:
Update setup.py

hariszaf committed:
Update setup_old.py

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update setup.py

hariszaf committed:
Update setup.py

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update test1.py

hariszaf committed:
Create test1_old.py

hariszaf committed:
Update test1.py

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update setup.py

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update volume_cb_hpolytope.cpp

hariszaf committed:
Update test1.py

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update test1.py

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Create .gitignore

hariszaf committed:
Delete test1_old.py

hariszaf committed:
Delete setup_old.py

hariszaf committed:
Delete bindings_old.h

hariszaf committed:
Delete bindings_old.cpp

hariszaf committed:
Delete volestipy_old.pyx

hariszaf committed:
merge with the GeomScale develop branch

Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Update bindings.h

hariszaf committed:
Update .gitignore

hariszaf committed:
Update volume_cb_hpolytope.cpp

hariszaf committed:
Update test1.py

hariszaf committed:
Update test2.py

hariszaf committed:
Update bindings.h

hariszaf committed:
Update setup.py

hariszaf committed:
Update credits.md

hariszaf committed:
on .gitignore: files on test/Testing/Temporary/ were removed from the .gitignore file

hariszaf committed:
first attempt - not working - for rounding

hariszaf committed:
rounding function: does not work; trying to figure it out

hariszaf committed:
round_val now works; still new_A and new_b need to be included

hariszaf committed:
Create metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update metabolic_net_pipeline.ipynb

hariszaf committed:
Update volestipy.pyx

hariszaf committed:
Update bindings.cpp

hariszaf committed:
Create test_rounding.py

hariszaf committed:
Create test1.py

hariszaf committed:
Create test2.py

hariszaf committed:
Delete test2.py

hariszaf committed:
Update test1.py

hariszaf committed:
Create metabolic_net_pipeline-checkpoint.ipynb

hariszaf committed:
Update setup.py

hariszaf committed:
Update .gitignore

hariszaf committed:
Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

hariszaf committed:
Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

Last update GeomScale volume_approximation before PR for rounding

hariszaf committed:
Merge branch 'develop' of https://github.com/GeomScale/volume_approximation into develop

hariszaf committed:
Delete test1.py

git rm test1.py

hariszaf committed:
Delete test2.py

git rm test2.py

hariszaf committed:
Update bindings.h

new class for the preprocessing and the get_full_dimensional_polytope() function in the bindings.h

hariszaf committed:
Update bindings.cpp

new class for the preprocessing and the get_full_dimensional_polytope() function in the bindings.cpp

hariszaf committed:
Update volestipy.pyx

new class for the preprocessing and the get_full_dimensional_polytope() function in the volestipy.pyx

hariszaf committed:
Update bindings.h

working on the lowDimHPolytopeCPP class - just a draft with the main ideas, not a stable version

hariszaf committed:
Update bindings.cpp

working on the lowDimHPolytopeCPP class - just a draft with the main ideas, not a stable version

hariszaf committed:
Update volestipy.pyx

working on the lowDimHPolytopeCPP class - just a draft with the main ideas, not a stable version

hariszaf committed:
Update bindings.h

set member variables for lowDimHPolytopeCPP class

hariszaf committed:
Update bindings.cpp

remove unessecary vars

hariszaf committed:
Update volestipy.pyx

remove unecessary vars in self.full_dim_polytope

hariszaf committed:
Update bindings.h

build the instructor for the full_dimensional_polytope() function

hariszaf committed:
Update bindings.cpp

1st draft for the get_full_dimensional_polytope() function

hariszaf committed:
Update volestipy.pyx

1st draft for the python interface of the get_full_dimensional_polytope() function

hariszaf committed:
Update bindings.h

using transpose matrices for N and A_full
removing the "extra" from the shift and b_full vectors as it is not needed

hariszaf committed:
Update bindings.cpp

2nd draft for the lowDimHPolytopeCPP class.
we changed the N and A_full matrices to transpose to feed them in the cpp file like this.
we removed the _extra from the vectors

hariszaf committed:
Update volestipy.pyx

2nd draft for the lowDimHPolytopeCPP class.
changed "_extra" vectors when needed.
first draft that actually compiles

hariszaf committed:
Create test_get_full_dim_polytope.py

test file for the get_full_dimensional_polytope function
not ready yet

hariszaf committed:
Update test_get_full_dim_polytope.py

changed the Aeq

hariszaf committed:
Update test_get_full_dim_polytope.py

final test for the get_full_dimensional_polytope() function

hariszaf committed:
Update bindings.cpp

print the N transpose and the full A transpose matrices from the C++ code

hariszaf committed:
Update volestipy.pyx

Attention! What C++ returns is the transpose matrix.
Thus, the [i,j] element of the full A is the [j,i] of what C++ returns. That is what we fixed here

hariszaf committed:
Update test_get_full_dim_polytope.py

print all the outputs of the get_full_dimensional_polytope() function

hariszaf committed:
Update bindings.cpp

remove test prints

hariszaf committed:
Update volestipy.pyx

remove test printing statements
add all the output in the return statement on the get_full_dim_polytope()
improved comments

hariszaf committed:
Delete metabolic_net_pipeline.ipynb

remove the jupyter notebook

hariszaf committed:
Delete metabolic_net_pipeline-checkpoint.ipynb

remove the hidden directory for the jupyter notebook

TolisChal commented on volestipy/volestipy/include/bindings.h:
```diff
@@ -27,51 +27,71 @@
+      MT A,Aeq;
+      VT b,beq;
+      
+      std::pair<Point,NT> CheBall;
```
Do you need CheBall in low dim polytope?

hariszaf replied:
Nope.
I will remove it.

TolisChal commented on volestipy/volestipy/volestipy.pyx:
```diff
@@ -115,3 +126,86 @@ cdef class HPolytope:
+      del A_full_extra_trans
+      
+      # finally, we need to build an HP object for the full dumensional polytope we got
+      full_dimensional_polytope = HPolytope(A_full,b_full)
```
Maybe del b_full after the declaration of object `full_dimensional_polytope`?

hariszaf replied:
If we remove ```b_full```, we should remove the ```A_full``` as well right?

TolisChal commented on volestipy/volestipy/volestipy.pyx:
```diff
@@ -115,3 +126,86 @@ cdef class HPolytope:
+      self._Aeq = Aeq
+      self._beq = beq
+      n_rows_of_A, n_cols_of_A = A.shape[0], A.shape[1] 
+      n_row_of_Aeq, n_cols_of_Aeq = Aeq.shape[0], Aeq.shape[1] 
```
Please check if:
1) n_rows_of_A == b.shape[0]
2) n_row_of_Aeq == beq.shape[0]
3) n_cols_of_A == n_cols_of_Aeq

If not raise an exception

hariszaf replied:
Three if statements were added to check for these equalities

hariszaf committed:
Update volestipy.pyx

Exceptions for the low_dim_polytope_cpp constructor added.
Deletion of the A_full and the b_full variables after the initialization of the object.

hariszaf committed:
Update bindings.h

CheBall was removed from the case of the lowDimHPolytopeCPP
